src/display/display_sdl.c

```diff
@@ -578,6 +578,23 @@ static void display_sdl_fullscreen()
 		SDL_ShowCursor(SDL_ENABLE);
 }
 
+/* Check if there is a Unicode TextInput event further down
+   the SDL queue. This is the only way we can distinguish AltGr
+   from plain right Alt, it appears; by the text the former
+   produces. */
+static int has_unicode_event_queued()
+{
+	SDL_Event outevent;
+	int pending_events = SDL_PeepEvents(&outevent,
+		1, SDL_PEEKEVENT, SDL_FIRSTEVENT, SDL_LASTEVENT);
+
+	if (pending_events < 1)
+		return 0;
+
+	return outevent.type == SDL_TEXTINPUT &&
+		!is_ascii_key(outevent.text.text[0]);
```

Comment:
If the user is moving the mouse or their CPU gets bogged down, there could be more than 1 event queued. We can deal with this using `SDL_PeepEvents` filtering and a larger buffer. Something like:

```suggestion
	const int MAX_EVENTS = 10;
	SDL_Event outevent[MAX_EVENTS];
	int pending_events = SDL_PeepEvents(outevent,
		MAX_EVENTS, SDL_PEEKEVENT, SDL_TEXTINPUT, SDL_TEXTINPUT);
	int i;
	for (i = 0; i < pending_events; i++) {
		if (!is_ascii_key(outevent[i].text.text[0])) {
			return true;
		}
	}
	return false;
```